flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -438,30 +446,35 @@ object GenerateUtils {
   def generateProctimeTimestamp(
       ctx: CodeGeneratorContext,
       contextTerm: String): GeneratedExpression = {
-    val resultTerm = ctx.addReusableLocalVariable("long", "result")
+    val resultType = new TimestampType(3)
+    val resultTypeTerm = primitiveTypeTermForType(resultType)
+    val resultTerm = ctx.addReusableLocalVariable(resultTypeTerm, "result")
     val resultCode =
       s"""
-         |$resultTerm = $contextTerm.timerService().currentProcessingTime();
+         |$resultTerm = $SQL_TIMESTAMP.fromEpochMillis(
+         |  $contextTerm.timerService().currentProcessingTime());
          |""".stripMargin.trim
     // the proctime has been materialized, so it's TIMESTAMP now, not PROCTIME_INDICATOR
-    GeneratedExpression(resultTerm, NEVER_NULL, resultCode, new TimestampType(3))
+    GeneratedExpression(resultTerm, NEVER_NULL, resultCode, resultType)
   }
 
   def generateCurrentTimestamp(
       ctx: CodeGeneratorContext): GeneratedExpression = {
     new CurrentTimePointCallGen(false).generate(ctx, Seq(), new TimestampType(3))
   }
 
-  def generateRowtimeAccess(
+  def generateTimestampAccess(
```

Comment:
Why change the method name? I think the original method describe the logic more accurate.
At the first glance at the new method name, I though it is used to access timestamp field from a row or record.